packages/repository/test/unit/repositories/atomic.repository.unit.ts

```diff
@@ -0,0 +1,351 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  Entity,
+  EntityNotFoundError,
+  juggler,
+  ModelDefinition,
+  DefaultAtomicCrudRepository,
+} from '../../..';
+
+describe('AtomicCrudRepository', () => {
+  let ds: juggler.DataSource;
+
+  class Note extends Entity {
+    static definition = new ModelDefinition({
+      name: 'Note',
+      properties: {
+        title: 'string',
+        content: 'string',
+        id: {name: 'id', type: 'number', id: true},
+      },
+    });
+
+    title?: string;
+    content?: string;
+    id: number;
+
+    constructor(data: Partial<Note>) {
+      super(data);
+    }
+  }
+
+  beforeEach(() => {
+    ds = new juggler.DataSource({
+      name: 'db',
+      connector: 'memory',
+    });
+  });
+
+  context('constructor', () => {
+    class ShoppingList extends Entity {
+      static definition = new ModelDefinition({
+        name: 'ShoppingList',
+        properties: {
+          id: {
+            type: 'number',
+            id: true,
+          },
+          created: {
+            type: () => Date,
+          },
+          toBuy: {
+            type: 'array',
+            itemType: 'string',
+          },
+          toVisit: {
+            type: Array,
+            itemType: () => String,
+          },
+        },
+      });
+
+      created: Date;
+      toBuy: String[];
+      toVisit: String[];
+    }
+
+    it('converts PropertyDefinition with array type', () => {
+      const originalPropertyDefinition = Object.assign(
+        {},
+        ShoppingList.definition.properties,
+      );
+      const listDefinition = new DefaultAtomicCrudRepository(ShoppingList, ds)
+        .modelClass.definition;
+      const jugglerPropertyDefinition = {
+        created: {type: Date},
+        toBuy: {
+          type: [String],
+        },
+        toVisit: {
+          type: [String],
+        },
+      };
+
+      expect(listDefinition.properties).to.containDeep(
+        jugglerPropertyDefinition,
+      );
+      expect(ShoppingList.definition.properties).to.containDeep(
+        originalPropertyDefinition,
+      );
+    });
+
+    it('throws if a connector instance is not defined for a datasource', () => {
+      const dsWithoutConnector = new juggler.DataSource({
+        name: 'ds2',
+      });
+      let repo;
+      expect(() => {
+        repo = new DefaultAtomicCrudRepository(
+          ShoppingList,
+          dsWithoutConnector,
+        );
+      }).to.throw(
+        /Connector instance must exist and support atomic operations/,
+      );
+    });
+
+    it('shares the backing PersistedModel across repo instances', () => {
```

Comment:
I copied the tests from DefaultCrudRepository unit test file because I was worried code coverage would drop if I didn't have them and since we extend from it.